Add .gitignore to preserve empty directories

ktomk opened the pull request:
The magento2-base package contains a few empty directories. The addition
of these files ensures those directories existence if magento2-base is
repackaged by a third party.

References:
- Issue: #3248
- Fixup of PR: #3550

piotrekkaminski commented:
please review @antonkril

piotrekkaminski commented:
Internal ID  MAGETWO-50039

vrann commented:
@kromk Thank you for the contribution!

Ignoring these directories from the git will make it hard to work with for the core developers. Having that, I propose to use .gitkeep file instead.

It is second by popularity method in this discussion:
http://stackoverflow.com/questions/115983/how-can-i-add-an-empty-directory-to-a-git-repository

ktomk commented:
How would that be different from the suggestion? Have problems to understand what is hard then for core developers, so maybe you can guide me.

vrann commented:
@ktomk sure, so .gitignore with the "*" in it makes the directory where it is located (app/code/Magento) ignored from the git, right? This is the directory where core developers write the code, so making it ignored will break the developers' workflow.

I suggest replacing it with the file which does not ignore anything and is used just to mark the directory as not empty.

ktomk commented:
@vrann: Thanks for the quick feedback. When I created the PR these folders were actually empty. If they should be kept under version control, just add a zero-byte `.gitignore` file, it will do the same but keep the files for core developers workflow as they have files under version control there (but not in this repository?! - I'm still a bit puzzled). Anyway that would be the option then.

I also reviewed the origin why I made this PR and from what I know there is no need for this any longer as re-packaging in Satis has been fixed for empty directories ca. a year ago or so. So maybe this change is not required any longer?

Also I should review why the commit does not apply clean any longer. This PR is over a year old, maybe this is just stale and should be reviewed in full again?

@hostep: Thanks for feedback as well.

vrann commented:
@hostep that's right but those directories should be ignored indeed. Though I'm fine with the empty ``.gitignore``, at some point we may decide to add something there.

@ktomk there are zero files on your installation because the Magento was most likely installed via composer and all the code goes to the vendor directory. But if sources retrieved from the GitHub, the files will be under app/code.

If this is not the issue anymore for the Satis then we can close PR. Otherwise please change to  zero-byte ``.gitignore``

ktomk committed:
Add .gitignore to preserve empty directories, on #3976

The magento2-base package contains a few empty directories. The addition
of these files ensures those directories existence if magento2-base is
repackaged by a third party.

NOTE: Commit is refreshed and updated for core devs, but it is not needed
      and should not be merged to upstream develop.

References:

- Issue: #3976

- Issue: #3248

- Fixup of PR: #3550

ktomk commented:
@vrann: Just reveiwed and refreshed, just spare this one. I close it, the leftovers are only for reference.
